### PR TITLE
DEP: Loosen dependency requirements

### PR DESCRIPTION
**Pull request by richford**
Following the same process as in this [groupyr PR](https://github.com/richford/groupyr/pull/46), this PR tidies up the dependency requirements. It removes unnecessary dependencies and widens the range for others. It also removes the deepdish dependency (it appears my beloved deepdish is no longer maintained and is not compatible with recent pandas versions) by copying the relevant parts into the new h5io module.

The same narrow restrictions on sklearn version mentioned in the groupyr PR above also apply here.

### CODE REVIEW

**Comment by richford** [comment]
Ugh, changed to WIP. I guess since I added in a lot of the deepdish.io functionality, I should write tests for those too.

**Comment by richford** [comment]
Okay, I transferred (with minor modifications) the unit tests over from dd.io. The total test coverage is still reduced but honestly, I have no plans to bring the h5io module up to 100%.

**Comment by richford** [comment]
@arokem, This is ready for review as long as you're willing to accept a -0.4% decrease in test coverage. I am.

File: afqinsight/h5io.py [hunk 1]
```diff
@@ -0,0 +1,728 @@
+"""Recursively save and load dictionaries to hdf5 files.
+
+This module copies extensively from deepdish.io
```
Comment (arokem) [inline afqinsight/h5io.py:3]: The deepdish [license](https://github.com/uchicago-cs/deepdish/blob/master/LICENSE) (BSD) requires that if you redistribute the code,  you reproduce the text of the license itself -- including the original copyright. Assuming most of the code in this file is either directly copied, or copied and then lightly modified, I would put it here in the module-level docstring.

File: afqinsight/tests/test_h5io.py [hunk 2]
```diff
@@ -0,0 +1,404 @@
+import numpy as np
```
Comment (arokem) [inline afqinsight/tests/test_h5io.py:1]: And same here

**Review by arokem — COMMENTED** [review]
Yes. This looks fine, but we need to include the original license (including the original copyright), assuming this is primarily code copied directly from the deepdish source.

File: afqinsight/h5io.py (hunk 1 above)
Comment (richford) [reply afqinsight/h5io.py]: Good catch. Thanks! Resolved in [22a155b](https://github.com/richford/AFQ-Insight/pull/66/commits/22a155bc3b4314aa0973f3c03da03e17c41fcc60).

File: afqinsight/tests/test_h5io.py (hunk 2 above)
Comment (richford) [reply afqinsight/tests/test_h5io.py:38]: Likewise, resolved in [22a155b](https://github.com/richford/AFQ-Insight/pull/66/commits/22a155bc3b4314aa0973f3c03da03e17c41fcc60).

**Pull request merged**